temporalcli/internal/printer/printer.go

```diff
@@ -0,0 +1,514 @@
+		}
+		if options.Table != nil {
+			if width := options.Table.FieldWidths[col.name]; width > 0 {
+				col.width = width
+			}
+			if align, ok := options.Table.FieldAlign[col.name]; ok {
+				col.align = align
+			}
+		}
+		adjusted = append(adjusted, col)
+	}
+	return adjusted
+}
+
+func (p *Printer) printTable(options *TableOptions, cols []*col, rows []map[string]colVal) {
+	if !options.NoHeader {
+		p.printHeader(cols)
+	}
+	p.printRows(cols, rows)
+}
+
+func (p *Printer) printHeader(cols []*col) {
+	colorer := p.TableHeaderColorer
+	if colorer == nil {
+		colorer = color.MagentaString
+	}
+	for _, col := range cols {
+		// We want to indent even the first field
+		p.writeStr("  ")
+		p.writeStr(tablewriter.Pad(colorer("%v", col.name), " ", col.width))
+	}
+	p.writeStr("\n")
+}
+
+func (p *Printer) printRows(cols []*col, rows []map[string]colVal) {
+	for _, row := range rows {
+		p.printRow(cols, row)
+	}
+}
+
+func (p *Printer) printRow(cols []*col, row map[string]colVal) {
+	for _, col := range cols {
+		// We want to indent even the first field
+		p.writeStr("  ")
+		p.printCol(col, row[col.name].text)
+	}
+	p.writeStr("\n")
+}
+
+func (p *Printer) printCol(col *col, data string) {
+	switch col.align {
+	case AlignCenter:
+		data = tablewriter.Pad(data, " ", col.width)
+	case AlignRight:
+		data = tablewriter.PadLeft(data, " ", col.width)
+	default:
+		data = tablewriter.PadRight(data, " ", col.width)
+	}
+	p.writeStr(data)
+}
+
+func (p *Printer) printCards(cols []*col, rows []map[string]colVal) {
+	for i, row := range rows {
+		// Extra newline between cards
+		if i > 0 {
+			p.writeStr("\n")
+		}
+		p.printCard(cols, row)
+	}
+}
+
+func (p *Printer) printCard(cols []*col, row map[string]colVal) {
+	nameValueRows := make([]map[string]colVal, 0, len(cols))
+	for _, col := range cols {
+		if !col.cardOmitEmpty || !reflect.ValueOf(row[col.name].val).IsZero() {
+			nameValueRows = append(nameValueRows, map[string]colVal{
+				"Name":  {val: col.name, text: col.name},
+				"Value": row[col.name],
+			})
+		}
+	}
+	nameValueCols := []*col{
+		{name: "Name"},
+		// We want to set the width to 1 here, because we want it to stretch as far
+		// as it needs to the right
+		{name: "Value", width: 1},
+	}
+	p.calculateUnsetColWidths(nameValueCols, nameValueRows)
+	p.printRows(nameValueCols, nameValueRows)
+}
+
+var jsonMarshalerType = reflect.TypeOf((*json.Marshaler)(nil)).Elem()
+
+func (p *Printer) textVal(v any) string {
+	ref := reflect.Indirect(reflect.ValueOf(v))
+	if ref.IsValid() && !ref.IsZero() && ref.Type() == reflect.TypeOf(time.Time{}) {
+		if p.FormatTime == nil {
+			return ref.Interface().(time.Time).Format(time.RFC3339)
+		}
+		return p.FormatTime(ref.Interface().(time.Time))
+	} else if ref.IsValid() && ((ref.Kind() == reflect.Struct && ref.CanInterface()) || ref.Type().Implements(jsonMarshalerType)) {
+		b, err := p.jsonVal(v, "", true)
+		if err != nil {
+			return fmt.Sprintf("<failed converting to string: %v>", err)
```

Comment:
For some reason I haven't fully-debugged, this error fires with any workflow result that includes a nested JSON object. For example, changing [this line](https://github.com/temporalio/samples-typescript/blob/9d980ee0c6e81a786d700746b5c6d6f0065b0ae5/hello-world/src/workflows.ts#L12) of the hello-world TypeScript sample to this:

```js
  return {greeting: {}};
```

Causes the command `./temporal workflow execute -w foo --type example -t hello-world -i '"Josh"'` to emit:

```
...
Results:
  RunTime  60ms
  Status   COMPLETED
  Result   <failed converting to string: json: error calling MarshalJSON for type json.RawMessage: invalid character '}' looking for beginning of value>
```

Reply:
This appears to be an issue with the payload shorthand in our Go API library. I have opened https://github.com/temporalio/api/issues/349.

Reply:
Tim made the fix and I updated the dependency. This should now work.